src/gui/plugins/mouse_drag/MouseDrag.cc

```diff
@@ -0,0 +1,791 @@
+      return;
+    }
+    this->rayQuery = this->scene->CreateRayQuery();
+
+    auto mat = this->scene->Material("Default/TransRed")->Clone();
+    mat->SetDepthCheckEnabled(false);
+    mat->SetDepthWriteEnabled(false);
+
+    this->arrowVisual = this->scene->CreateArrowVisual();
+    this->arrowVisual->SetMaterial(mat);
+    this->arrowVisual->ShowArrowHead(true);
+    this->arrowVisual->ShowArrowShaft(true);
+    this->arrowVisual->ShowArrowRotation(false);
+
+    this->boxVisual = scene->CreateVisual();
+    this->boxVisual->AddGeometry(scene->CreateBox());
+    this->boxVisual->SetInheritScale(false);
+    this->boxVisual->SetMaterial(mat);
+    this->boxVisual->SetUserData("gui-only", static_cast<bool>(true));
+
+    auto planeMat = this->scene->Material("trans-gray");
+    if (!planeMat)
+    {
+      planeMat = this->scene->CreateMaterial("trans-gray");
+      planeMat->SetAmbient(1.0, 1.0, 1.0);
+      planeMat->SetDiffuse(1.0, 1.0, 1.0);
+      planeMat->SetSpecular(1.0, 1.0, 1.0);
+      planeMat->SetTransparency(0.7);
+      planeMat->SetCastShadows(false);
+      planeMat->SetReceiveShadows(false);
+      planeMat->SetLightingEnabled(false);
+    }
+
+    this->planeVisual = scene->CreateVisual();
+    this->planeVisual->AddGeometry(scene->CreatePlane());
+    this->planeVisual->SetInheritScale(false);
+    this->planeVisual->SetMaterial(planeMat);
+    this->planeVisual->SetUserData("gui-only", static_cast<bool>(true));
+  }
+
+  std::lock_guard<std::mutex> lock(this->mutex);
+
+  // Update the visualization
+  if (!this->dragActive)
+  {
+    this->arrowVisual->SetVisible(false);
+    this->boxVisual->SetVisible(false);
+    this->planeVisual->SetVisible(false);
+  }
+  else if (this->mode == MouseDragMode::ROTATE)
+  {
+    this->boxVisual->SetLocalPosition(this->applicationPoint);
+    this->boxVisual->SetLocalRotation(this->goalRot);
+    this->boxVisual->SetLocalScale(1.2 * this->bboxSize);
+
+    this->arrowVisual->SetVisible(false);
+    this->boxVisual->SetVisible(true);
+    this->planeVisual->SetVisible(false);
+  }
+  else if (this->mode == MouseDragMode::TRANSLATE)
+  {
+    math::Vector3d axisDir = this->target - this->applicationPoint;
+    math::Quaterniond quat;
+    quat.SetFrom2Axes(math::Vector3d::UnitZ, axisDir);
+    double scale = 2 * this->bboxSize.Length();
+    this->arrowVisual->SetLocalPosition(this->applicationPoint);
+    this->arrowVisual->SetLocalRotation(quat);
+    this->arrowVisual->SetLocalScale(scale, scale, axisDir.Length() / 0.75);
+
+    quat.SetFrom2Axes(math::Vector3d::UnitZ, -this->plane.Normal());
+    scale = this->applicationPoint.Distance(this->camera->WorldPose().Pos());
+    this->planeVisual->SetLocalRotation(quat);
+    this->planeVisual->SetLocalPosition(this->applicationPoint);
+    this->planeVisual->SetLocalScale(scale);
+
+    this->arrowVisual->SetVisible(true);
+    this->boxVisual->SetVisible(false);
+    this->planeVisual->SetVisible(true);
```

Comment:
Can we make this configurable? After playing with it for a while, I'm not sure if it's helping a whole lot. What do you think?

Reply:
I made the plane always visible in 78f86a5, do you think it's more intuitive? Honestly, I'm not loving it either and might just remove it